pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -57,14 +59,19 @@ type AppHandler struct {
 // Dispatch apply manifests into k8s.
 func (h *AppHandler) Dispatch(ctx context.Context, manifests ...*unstructured.Unstructured) error {
 	h.initDispatcher()
-	_, err := h.dispatcher.Dispatch(ctx, manifests)
+	rctx := common2.NewReconcileContext(ctx, types2.NamespacedName{
```

Comment:
Why create a new ctx for Dispatch() but re-use ReconcileContext passed in DispatchAndGC() ?

Reply:
A type check should be there. I will fix it.